### PR TITLE
bitnami/spring-cloud-data-flow Fix for #13383

### LINKED ISSUE #13383: [bitnami/spring-cloud-data-flow] Unable to set SPRING_APPLICATION_JSON env var
### Name and Version

bitnami/spring-cloud-data-flow 2.9.6

### What steps will reproduce the bug?

As per [Using SPRING_APPLICATION_JSON](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#_using_code_spring_application_json_code), you can use a `SPRING_APPLICATION_JSON` environment variable to set Data Flow server properties.

However when setting on a values.yaml file something like:
```
server:
[...]
  extraEnvVars:
    - name: SPRING_APPLICATION_JSON
      value: { "maven": { "remote-repositories":...
```

the resulting template yields something like:
```
[...]
            - name: SPRING_APPLICATION_JSON
              value: "map[maven:map[remote-repositories:map[...
[...]
```

### Are you using any custom parameters or values?

on a `values.yaml` file:
```
server:
[...]
  extraEnvVars:
    - name: SPRING_APPLICATION_JSON
      value: { "maven": { "remote-repositories":...
```
(would be nice to pass this var through an exisiting env variable, like `--set server.extraVars[0].SPRING_APPLICATION_JSON=$MY_ENV_VAR` but haven't found a way to do so)

### What is the expected behavior?

```
[...]
            - name: SPRING_APPLICATION_JSON
              value: {"maven":{"remote-repositories":{...
```

### What do you see instead?

```
[...]
            - name: SPRING_APPLICATION_JSON
              value: "map[maven:map[remote-repositories:map[...
[...]
```

### Additional information

changing range on line 172 on `deployment.yaml` to 

```
            {{- range $key, $value := .Values.server.extraEnvVars }}
            - name: {{ $value.name }}
              {{- if eq $value.name "SPRING_APPLICATION_JSON" }}
              value: {{ $value.value | toJson | quote }}
              {{- else }}
              value: "{{ $value.value }}"
              {{- end }}
            {{- end }}
```

seems enough to fix this. Another workaorund to manually patch the file is to extend bitnami's SCDF Docker image and add the aforementioned variable

### PR DESCRIPTION
**Pull request by juanpablo-santos**
Signed-off-by: Juan Pablo Santos Rodríguez <juanpablo.santos@gmail.com>

### Description of the change
Provides a Fix for #13383, _Unable to set SPRING_APPLICATION_JSON env var_

### Benefits

will be able to set `SPRING_APPLICATION_JSON` through a values.yaml file

### Applicable issues

  - fixes #13383

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @juanpablo-santos,

Thank you very much for your contribution, can you please take a look at my comment?

File: bitnami/spring-cloud-dataflow/templates/server/deployment.yaml [hunk 1]
```diff
@@ -171,7 +171,11 @@ spec:
               value: 'docker://{{ include "common.images.image" (dict "imageRoot" .Values.server.composedTaskRunner.image) }}'
             {{- range $key, $value := .Values.server.extraEnvVars }}
             - name: {{ $value.name }}
+              {{- if eq $value.name "SPRING_APPLICATION_JSON" }}
+              value: {{ $value.value | toJson | quote }}
+              {{- else }}
               value: "{{ $value.value }}"
+              {{- end }}
             {{- end }}
```
[the remark this answers is not in the document]
Comment (juanpablo-santos) [reply bitnami/spring-cloud-dataflow/templates/server/deployment.yaml]: Hi @migruiz4 , thanks for your review :-) I've applied the suggested fix, so it should be ready to go

[the remark this answers is not in the document]
Comment (juanpablo-santos) [reply bitnami/spring-cloud-dataflow/templates/server/deployment.yaml]: Also, just noticed that one of the cicd checks has failed after last commit, but not sure why. Surely the stack trace is self explanatory, but I'm relatively new to helm, so no clue :-?

**Comment by juanpablo-santos** [comment]
closing this PR as it's no longer needed, #13383 already provides a solution for this.

**Pull request closed without merging**